ambry-replication/src/test/java/com.github.ambry.replication/CloudToStoreReplicationManagerTest.java

```diff
@@ -0,0 +1,210 @@
+        return null;
+      }
+    };
+    // create vcr node
+    List<Port> vcrPortList = Arrays.asList(new Port(12310, PortType.PLAINTEXT), new Port(12410, PortType.SSL));
+    vcrNode = new MockDataNodeId("localhost", vcrPortList, Collections.singletonList(VCR_MOUNT_PATH), CLOUD_DC_NAME);
+    clusterMap = new MockClusterMap();
+    currentNode = clusterMap.getDataNodeIds().get(0);
+    mockClusterSpectator = new MockClusterSpectator(Collections.singletonList(vcrNode));
+    long replicaCapacity = clusterMap.getAllPartitionIds(null).get(0).getReplicaIds().get(0).getCapacityInBytes();
+    Properties properties = new Properties();
+    properties.setProperty("store.segment.size.in.bytes", Long.toString(replicaCapacity / 2L));
+    properties.setProperty("clustermap.cluster.name", "test");
+    properties.setProperty("clustermap.datacenter.name", "DC1");
+    properties.setProperty("clustermap.vcr.datacenter.name", "DC1");
+    properties.setProperty("clustermap.host.name", "localhost");
+    properties.setProperty("clustermap.dcs.zk.connect.strings", zkJson.toString(2));
+    properties.setProperty("replication.cloud.token.factory", "com.github.ambry.replication.MockFindTokenFactory");
+    properties.setProperty("disk.manager.enable.segment.pooling", "true");
+    verifiableProperties = new VerifiableProperties(properties);
+    storeConfig = new StoreConfig(verifiableProperties);
+    clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+    replicationConfig = new ReplicationConfig(verifiableProperties);
+    serverConfig = new ServerConfig(verifiableProperties);
+    mockHelixParticipant = new MockHelixParticipant(clusterMapConfig);
+  }
+
+  /**
+   * Test both success and failure cases when adding cloud replica
+   * @throws Exception
+   */
+  @Test
+  public void cloudReplicaAdditionTest() throws Exception {
+    StorageManager storageManager =
+        new StorageManager(storeConfig, new DiskManagerConfig(verifiableProperties), Utils.newScheduler(1, true),
+            clusterMap.getMetricRegistry(), null, clusterMap, currentNode, null, mockHelixParticipant, new MockTime(),
+            null);
+    CloudToStoreReplicationManager cloudToStoreReplicationManager =
+        new CloudToStoreReplicationManager(replicationConfig, clusterMapConfig, storeConfig, storageManager,
+            storeKeyFactory, clusterMap, mockScheduler, currentNode, null, clusterMap.getMetricRegistry(), null,
+            storeKeyConverterFactory, serverConfig.serverMessageTransformer, mockClusterSpectator,
+            mockHelixParticipant);
+    storageManager.start();
+    cloudToStoreReplicationManager.start();
+    mockClusterSpectator.spectate();
+    // 1. test adding cloud replica that is not present locally
+    mockHelixParticipant.onPartitionBecomeLeaderFromStandby(NEW_PARTITION_NAME);
+    assertNull("Thread pool for DC1 should be null",
+        cloudToStoreReplicationManager.getReplicaThreadPoolByDc().get("DC1"));
```

Comment:
I don't think we should expose an internal thread pool just for the test to check if an operation succeeded.  Perhaps we can expose a more specific method like hasStoreForDatacenter() in the ReplicationEngine base class?